platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,200 @@
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static String entityID;
+  private static EntityInformation testInfo;
+
+  private static String entityID2;
+  private static EntityInformation testInfo2;
+
+  private static final String METADATA_FILE = "SPMetadata";
+  private static final String METADATA_FILE_2 = "SPMetadata2";
+
+  private static final String CERTIFICATE_PATH =
+      Paths.get("src", "test", "resources", "Certificate.cert").toString();
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml/test";
+  private static final String ENTITY_ID_2 =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static String certificate;
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read Certificate file into certificate
+    try (Stream<String> lines = Files.lines(Paths.get(CERTIFICATE_PATH))) {
+      certificate = lines.collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    // read SPMetadata file into spMetadata
+    List<String> spMetadata = new ArrayList<>();
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE)));
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE_2)));
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+
+    // set up metadata
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+
+    // get the first metadata
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+    entityID = testEntry.getKey();
+    testInfo = testEntry.getValue();
+
+    testEntry = it.next();
+    entityID2 = testEntry.getKey();
+    testInfo2 = testEntry.getValue();
+
+    // get the second metadata
+
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat(testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void testentityID2s() throws Exception {
+    assertThat(entityID2, is(equalTo(ENTITY_ID)));
+    assertThat(entityID, is(equalTo(ENTITY_ID_2)));
```

Comment:
✏️  You should rename `ENTITY_ID` and `ENTITY_ID_2` cause it's confusing when you assert that `entityID2` is equal to `ENTITY_ID` on lines 114 and 115